horace_core/sqw/@sqw/mask.m

```diff
@@ -49,8 +49,8 @@
 % Section the pix array, if sqw type, and update urange
 if is_sqw_type(win)
     mask_pix = logical(replicate_array (mask_array, win.data.npix));
-    wout.data.pix=[];   % Clear the memory of a large array that is going to be replaced - but is a field, so musst leave present
-    wout.data.pix=win.data.pix(:,mask_pix);
+    wout.data.pix=PixelData();   % Clear the memory of a large array that is going to be replaced - but is a field, so musst leave present
```

Comment:
does this assignment still free the previously assigned memory?

Reply:
Yes, I think so. The old array gets dereferenced, so it'll be deleted